### PR TITLE
[NEW] Add author name in the info messages

### LINKED ISSUE #1689: missing user name for status messages, like 'Has joined channel', 'Message removed' etc.
- Your Rocket.Chat Experimental app version: 4.4.0 (and 4.3.1 stable version)
- Your Rocket.Chat server version: 2.4.5
- Device (or Simulator) you're running with: OnePlus 7 Pro (Android 10.0.3)

It would be helpful if the user name was displayed in the status messages, such as 'Has joined channel', 'Message removed'. In their current form, the status messages are not very informative.

rc

### PR DESCRIPTION
**Pull request by youssef-md**
@RocketChat/ReactNative

Closes #1689

In this PR was done:
* Added the name of the author of the action(join a room, delete a message)
* Limited his name so that the entire info message stays in a single line

ezgif com-optimize

### CODE REVIEW

File: app/containers/message/styles.js [hunk 1]
```diff
@@ -161,5 +161,9 @@ export default StyleSheet.create({
 	},
 	readReceipt: {
 		lineHeight: 20
+	},
+	username: {
+		marginLeft: 8,
+		color: '#333'
```
Comment (djorkaeffalexandre) [inline app/containers/message/styles.js]: Never use constant colors on `Text`, you need to set themed colors on inline style.

Comment (djorkaeffalexandre) [inline app/containers/message/styles.js]: Add `sharedStyles` with some `text*`.

File: app/containers/message/Message.js [hunk 2]
```diff
@@ -63,11 +63,15 @@ MessageInner.displayName = 'MessageInner';
 const Message = React.memo((props) => {
 	if (props.isThreadReply || props.isThreadSequential || props.isInfo) {
 		const thread = props.isThreadReply ? <RepliedThread {...props} /> : null;
+		const { username } = props.author;
```
Comment (djorkaeffalexandre) [inline app/containers/message/Message.js]: We'll have author always?

File: app/containers/message/Message.js [hunk 3]
```diff
@@ -63,11 +63,15 @@ MessageInner.displayName = 'MessageInner';
...
 		return (
 			<View style={[styles.container, props.style]}>
 				{thread}
 				<View style={[styles.flex, styles.center]}>
 					<MessageAvatar small {...props} />
+					<Text style={styles.username}>
```
Comment (djorkaeffalexandre) [inline app/containers/message/Message.js]: Try `flex: 1` or set `width` of text and it'll `ellipsis` programmatically, you can remove this logic with substring.

Comment (djorkaeffalexandre) [inline app/containers/message/Message.js]: `style={[styles.username, { color: themes[theme].color }]}`

Comment (youssef-md) [reply app/containers/message/Message.js]: My bad, I've chosen the titleText

**Comment by localguru** [comment]
Hi, does the PR only publish the username? What if on server side "full name" is activated, then it should be `Firstname Lastname @username Timestamp has xyz the channel`. I also think there should be no difference in content between the app and browser/desktop client.

**Comment by diegolmello** [comment]
@localguru Mobile has a very smaller space to show all those info without breaking line.
I'll talk to the design team about this.
Thanks.

File: app/containers/message/Message.js (hunk 3 above)
Comment (youssef-md) [reply app/containers/message/Message.js]: Thanks for the advice, by limiting the text width and setting `numberOfLines` to 1 all the info messages are know aligned due to the author name having the same size.

**Comment by dkoo761** [comment]
Just launched my first Rocket.Chat community today (🎉🎉🎉) and noticed this issue right away as soon as people started signing up :)

Is there a reason to try and keep it to one line? It seems like the way it works on the mobile web UI looks fine with it wrapping to 2 lines most of the time. I would prefer to keep all the info (avatar, full name, username, time joined, joined message) and have it wrap to 2 lines, rather than truncating the info.

**Comment by diegolmello** [comment]
> Just launched my first Rocket.Chat community today (🎉🎉🎉) and noticed this issue right away as soon as people started signing up :)
> 
> Is there a reason to try and keep it to one line? It seems like the way it works on the mobile web UI looks fine with it wrapping to 2 lines most of the time. I would prefer to keep all the info (avatar, full name, username, time joined, joined message) and have it wrap to 2 lines, rather than truncating the info.

Congrats on your community, man!
Yeah, we hold this PR just because we needed to tweak it a bit, but we've never found time to do it.
Hopefully we can take a look at it soon.

**Comment by dkoo761** [comment]
@diegolmello Thanks! I'm really excited to be building a community on Rocket.Chat.

I've built many Meteor apps (been using it full-time since 2013) and also have experience building React Native apps for iOS and Android. How about I take a crack at fixing this up as my first Rocket.Chat issue? :)

**Comment by diegolmello** [comment]
@dkoo761 Sure! Go ahead :)

**Comment by dkoo761** [comment]
Great, thanks!

**Comment by dkoo761** [comment]
@diegolmello I've got a local Rocket.Chat dev server environment up and running and am having a look through the code there before moving onto the React Native code. I've noticed that it takes about 4-5 minutes to restart after each code change. Is that normal? Seems like it would make development extremely slow. Could you share how you guys deal with that? Maybe you comment out imports or code you're not using for a specific feature or something?

Also, is there a better place to be asking general dev questions like this? I've tried asking on the forums and on open.rocket.chat but didn't get any reply on either one.

**Comment by dkoo761** [comment]
I did some profiling and made a list of everything that took more than 1 second:

(#3) Total: 104,696 ms (Build App)

| Top leaves:
| files.readFile..........................................16,258 ms (47956)
| other ImportScanner#findImportedModuleIdentifiers.......14,636 ms (6832)
| files.stat..............................................12,892 ms (91138)
| other minifyCssFiles.....................................8,229 ms (2)
| Babel.compile............................................5,001 ms (4291)
| other compileUnibuild (the app)..........................3,391 ms (3)
| other ImportScanner#resolve..............................3,169 ms (30116)
| files.watchFile..........................................2,857 ms (174052)
| other plugin templating-compiler.........................2,593 ms (2)
| linker File#getPrelinkedOutput...........................2,496 ms (4765)
| getPrelinkedFiles toStringWithSourceMap..................2,358 ms (3)
| other ImportScanner#scanImports for the app..............2,239 ms (3)
| files.writeFile..........................................1,844 ms (6547)
| other safeWatcher.watch..................................1,756 ms (172508)
| plugin rocketchat:version................................1,577 ms (3)
| other PackageSource#_findSources for os..................1,541 ms (1)
| files.readdirWithTypes...................................1,498 ms (49851)
| CssTools.stringifyCss....................................1,385 ms (2)
| CssTools.parseCss........................................1,319 ms (124)
| other bundler writeSiteArchive...........................1,046 ms (1)

Total: 250,253 ms (Server startup)

I20201007-15:26:12.951(-7)? | Top leaves:
I20201007-15:26:14.532(-7)? | Meteor.startup at module (app/authorization/server/startup.js:9:8).101,713 ms (1)
I20201007-15:26:14.532(-7)? | other require("/app/lib/server/startup/settings.js")....35,951 ms (1)
I20201007-15:26:14.532(-7)? | Meteor.startup at module (app/autolinker/server/settings.js:5:8).7,488 ms (1)
I20201007-15:26:14.532(-7)? | Meteor.startup at module (app/livechat/server/config.js:5:8).6,207 ms (1)
I20201007-15:26:14.533(-7)? | require("/node_modules/@rocket.chat/apps-engine/server/AppManager.js").3,139 ms (1)
I20201007-15:26:14.533(-7)? | Meteor.startup at module (app/meteor-accounts-saml/server/startup.ts:21:8).2,925 ms (1)
I20201007-15:26:14.533(-7)? | Meteor.startup at module (app/autotranslate/server/settings.js:5:8).2,418 ms (1)
I20201007-15:26:14.533(-7)? | Meteor.startup at module (app/markdown/server/settings.js:5:8).2,413 ms (1)
I20201007-15:26:14.533(-7)? | other require("/app/models/server/models/_oplogHandle.ts").2,076 ms (1)
I20201007-15:26:14.533(-7)? | other require("/app/models/server/models/Subscriptions.js").1,905 ms (1)
I20201007-15:26:14.541(-7)? | Meteor.startup at module (app/videobridge/server/settings.js:5:8).1,737 ms (1)
I20201007-15:26:14.541(-7)? | other require("/app/models/server/models/Messages.js")...1,732 ms (1)
I20201007-15:26:14.541(-7)? | Meteor.startup at module (app/sms/server/settings.js:5:8).1,559 ms (1)
I20201007-15:26:14.541(-7)? | other require("/app/assets/server/assets.js")............1,553 ms (1)
I20201007-15:26:14.541(-7)? | Meteor.startup at module (app/search/server/service/providerService.js:123:8).1,368 ms (1)
I20201007-15:26:14.541(-7)? | other require("/node_modules/meteor/accounts-base/server_main.js").1,245 ms (1)
I20201007-15:26:14.543(-7)? | Meteor.startup at module (app/authentication/server/startup/settings.ts:5:8).1,218 ms (1)
I20201007-15:26:14.543(-7)? | Meteor.startup at module (app/google-vision/server/settings.js:5:8).1,210 ms (1)
I20201007-15:26:14.543(-7)? | Meteor.startup at module (app/cas/server/cas_rocketchat.js:9:8).1,206 ms (1)
I20201007-15:26:14.543(-7)? | other require("/app/models/server/models/Rooms.js")......1,127 ms (1)
I20201007-15:26:14.543(-7)? | Meteor.startup at module (app/crowd/server/settings.js:5:8).1,114 ms (1)

Again my apologies, I know this isn't the right place to be having this discussion, but it's currently blocking me from getting started with development. So just hoping you can point me in the right direction.

**Comment by diegolmello** [comment]
@dkoo761 you can ask for help on https://open.rocket.chat/channel/support or https://open.rocket.chat/channel/dev

**Comment by dkoo761** [comment]
@diegolmello Perfect, I'll ask there. Thanks so much!

**Comment by dkoo761** [comment]
Just submitted my PR for this ^^^

**Pull request closed without merging**